core/src/main/scala/io/qbeast/model/CubeWeights.scala

```diff
@@ -56,21 +62,21 @@ class CubeWeightsBuilder(
       while (continue && containers.hasNext) {
         val cubeId = containers.next()
         val weightAndCount = weights.getOrElseUpdate(cubeId, new WeightAndCount(MaxValue, 0))
-        if (weightAndCount.count < partitionedDesiredCubeSize) {
+        if (weightAndCount.count < boostSize) {
           weightAndCount.count += 1
-          if (weightAndCount.count == partitionedDesiredCubeSize) {
+          if (weightAndCount.count >= boostSize) {
             weightAndCount.weight = weight
           }
           continue = announcedSet.contains(cubeId) || replicatedSet.contains(cubeId)
         }
       }
     }
     weights.map {
-      case (cubeId, weightAndCount) if weightAndCount.count == partitionedDesiredCubeSize =>
-        val nw = NormalizedWeight(weightAndCount.weight)
+      case (cubeId, weightAndCount) if weightAndCount.count == boostSize =>
+        val nw = NormalizedWeight(weightAndCount.weight) * boostSize
```

Comment:
Solved in 010ae95e95b9c8d9838210c23d9088b8081e89a2 

It should not be `boostSize`, but `desiredSize/boostSize` proportion. Also, the other case should initialize `NormalizedWeight` with `desiredSize` instead of `boostSize`